Client/mods/deathmatch/logic/luadefs/CLuaVectorGraphicDefs.cpp

```diff
@@ -0,0 +1,318 @@
+                                return false;
+
+                            if (!pFile->Load(pParentResource, CScriptFile::MODE_READ))
+                            {
+                                delete pVectorGraphic;
+                                delete pFile;
+
+                                m_pScriptDebugging->LogCustom(luaVM, SString("Unable to load SVG (file doesn't exist) [%s]", path.c_str()));
+                                return false;
+                            }
+                            else
+                            {
+                                SString strXmlData;
+                                pFile->Read(pFile->GetSize(), strXmlData);
+
+                                // We don't need the file handler anymore
+                                delete pFile;
+
+                                if (strXmlData.empty() || !pVectorGraphic->LoadFromString(strXmlData))
+                                {
+                                    delete pVectorGraphic;
+
+                                    m_pScriptDebugging->LogCustom(luaVM, "Unable to load SVG file (check for XML syntax errors)");
+                                    return false;
+                                }
+                            }
+
+                            return true;
+                        },
+                        [funcRef](const bool isLoaded) {
+                            CLuaMain* pLuaMain = m_pLuaManager->GetVirtualMachine(funcRef.GetLuaVM());
+                            if (pLuaMain)
+                            {
+                                CLuaArguments arguments;
+                                arguments.PushBoolean(isLoaded);
+                                arguments.Call(pLuaMain, funcRef);
+                            }
+                        });
+                }
+                else
+                {
+                    if (!pFile->Load(pParentResource, CScriptFile::MODE_READ))
+                    {
+                        delete pVectorGraphic;
+                        delete pFile;
+
+                        m_pScriptDebugging->LogCustom(luaVM, SString("Unable to load SVG (file doesn't exist) [%s]", pathOrRawData.value().c_str()));
+                        return false;
+                    }
+                    else
+                    {
+                        SString strXmlData;
+                        pFile->Read(pFile->GetSize(), strXmlData);
+
+                        // We don't need the file handler anymore
+                        delete pFile;
+
+                        if (strXmlData.empty() || !pVectorGraphic->LoadFromString(strXmlData))
+                        {
+                            delete pVectorGraphic;
+
+                            m_pScriptDebugging->LogCustom(luaVM, "Unable to load SVG file (check for XML syntax errors)");
+                            return false;
+                        }
+                    }
+                }
```

Comment:
The code is currently duplicated twice, once for the callback and once for the non-callback. I would like this to be refactored into a e.g. single private function call.